src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java

```diff
@@ -175,6 +179,106 @@ public void testDBSchema() throws SQLException {
+                }
+            }
+
+            final MessageFormat atLeastOneFoundFormat = new MessageFormat(TestResource.getResource("R_atLeastOneFound"));
+            assertTrue(hasResults, atLeastOneFoundFormat.format(schemaMsgArgs));
+
+            final MessageFormat dashCatalogFormat = new MessageFormat(TestResource.getResource("R_atLeastOneFound"));
+            assertTrue(hasDashCatalogSchema, dashCatalogFormat.format(new Object[] {testSchema}));
+        } finally {
+            if (dropDatabase) {
+                connection.createStatement().execute(String.format("DROP DATABASE [%s]", testCatalog));
+            }
+        }
+    }
+
+    /**
+     * Tests that the catalog parameter containing - is escaped by
+     * {@link SQLServerDatabaseMetaData#getSchemas(String catalog, String schemaPattern)}.
+     * @throws SQLException
+     */
+    @Test
+    public void testDBSchemasForDashedCatalogNameWithPattern() throws SQLException {
+        final UUID id = UUID.randomUUID();
+        final String testCatalog = "dash-catalog"+id;
+        final String testSchema = "some-schema"+id;
+        boolean dropDatabase = false;
+
+        try (final Connection dashConn = DriverManager.getConnection(connectionString);
+             final Statement dashStatement = dashConn.createStatement()) {
+
+            connection.createStatement().execute(String.format("CREATE DATABASE [%s]", testCatalog));
+            dropDatabase = true;
+            dashStatement.execute(String.format("USE [%s]", testCatalog));
+            dashStatement.execute(String.format("CREATE SCHEMA [%s]", testSchema));
+
+            final DatabaseMetaData databaseMetaData = connection.getMetaData();
+            final ResultSet rs = databaseMetaData.getSchemas(testCatalog, "some-%");
+
+            final MessageFormat schemaEmptyFormat = new MessageFormat(TestResource.getResource("R_nameEmpty"));
+            final Object[] schemaMsgArgs = {testSchema};
+            final Object[] catalogMsgArgs = {testCatalog};
+
+            boolean hasResults = false;
+            while (rs.next()) {
+                hasResults = true;
+                final String schemaName = rs.getString(1);
+                final String catalogName = rs.getString(2);
+                assertTrue(!StringUtils.isEmpty(schemaName), schemaEmptyFormat.format(schemaMsgArgs));
+                assertTrue(!StringUtils.isEmpty(catalogName), schemaEmptyFormat.format(catalogMsgArgs));
+                assertEquals(schemaName, schemaMsgArgs[0]);
+                assertEquals(catalogName, catalogMsgArgs[0]);
+            }
+
+            final MessageFormat atLeastOneFoundFormat = new MessageFormat(TestResource.getResource("R_atLeastOneFound"));
+            assertTrue(hasResults, atLeastOneFoundFormat.format(schemaMsgArgs));
+        } finally {
+            if (dropDatabase) {
+                connection.createStatement().execute(String.format("DROP DATABASE [%s]", testCatalog));
```

Comment:
You're using the generic connection and creating statement, which is not closed (same for other method), you could use _Utils_ functions to drop database for consistency and that clear up objects as well.